sqlglot/optimizer/scope.py

```diff
@@ -711,6 +711,31 @@ def _traverse_subqueries(scope):
         scope.subquery_scopes.append(top)
 
 
+def _traverse_udtfs(scope):
+    sources = {}
+
+    if isinstance(scope.expression, exp.Unnest):
```

Comment:
why are you checking specifically for unnest?

Reply:
I'll work on handling `exp.Lateral` too. I think these two are the only meaningful `UDTF`s to deal with here, as the other one is `Values`.